packages/abi-coder/src/function-fragment.ts

```diff
@@ -66,6 +67,21 @@ export class FunctionFragment<
     return this.jsonFn.inputs.length > 1 || isPointerType(inputTypes[0]?.type || '');
   }
 
+  isOutputDataHeap(): boolean {
+    const outputType = this.jsonAbi.types.find((t) => t.typeId === this.jsonFn.output.type);
```

Comment:
we have the `findOrThrow` utility to get rid of the `undefined` in `T | undefined`. It should throw anyways if it doesn't find the type.